source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/teamexplorer/internal/TeamExplorerHelpers.java

```diff
@@ -279,4 +311,219 @@ public static IViewPart showView(final String viewID) {
 
         return view;
     }
+
+    public static boolean tryParseProtocolHandlerArguments(final String[] applicationArgs) {
+
+        if (applicationArgs == null) {
+            return false;
+        }
+
+        boolean found = false;
+
+        for (final String arg : applicationArgs) {
```

Comment:
You might want to add comments here on how you are paring the url because without them this is confusing to follow